### PR TITLE
feat(tagging): Add LocationProvider component.

### PR DESCRIPTION
**Pull request by tajespasarela**
EX-5092

This PR is to add the new `LocationProvider` component to the Archetype.
Also, the structure of the components was refactored a little bit to eliminate repetition.

## Motivation and context
The idea is to have the default locations configured and ready for the query tagging, and for future results events tagging.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?

Executed a search from each different components, and see the tagging call in the devtools with the right `feature:location`

### CODE REVIEW

File: src/components/results/recommendations.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div
-    v-if="(!$x.query.search || $x.noResults) && $x.recommendations.length > 0"
```
Comment (pmareke) [inline src/components/results/recommendations.vue:3]: What happened here? When did the recommendations appearance condition change?

**Review by pmareke — APPROVED** [review]

File: src/components/results/recommendations.vue (hunk 1 above)
Comment (tajespasarela) [reply src/components/results/recommendations.vue:3]: they were moved to the main.vue component

File: src/components/desktop/desktop.vue [hunk 2]
```diff
@@ -11,7 +11,9 @@
         class="x-list x-list--vertical x-list--gap-05 x-list--align-stretch x-list__item--expand"
       >
         <SearchBox />
-        <RelatedTags v-if="!$x.isEmpathizeOpen && $x.relatedTags.length > 0" />
+        <LocationProvider location="predictive_layer">
```
Comment (javieri-empathy) [inline src/components/desktop/desktop.vue:14]: Should the search-box be also contained in the predictive layer? Right now it sends `none` as location.

File: src/components/main.vue [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        <div
+          v-if="$x.totalResults === 0 || $x.spellcheckedQuery"
+          class="x-list x-list--padding-06 x-list--padding-bottom"
+        >
+          <SpellcheckMessage />
+          <NoResultsMessage />
+        </div>
+      </div>
+
+      <Results />
```
Comment (javieri-empathy) [inline src/components/main.vue:16]: Not for this PR, but something to take into account probably in the track pending query task. If you scroll fast tracking is skipped until you stop scrolling due to the debounce.

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: src/components/main.vue (hunk 3 above)
Comment (tajespasarela) [reply src/components/main.vue:16]: noted there https://searchbroker.atlassian.net/browse/EX-4962

File: src/components/main.vue [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+      <PartialResults />
+
+      <LocationProvider v-if="!$x.query.search" location="no_results">
```
[the remark this answers is not in the document]
Comment (tajespasarela) [reply src/components/main.vue]: This is what happens when I copy&paste. Good catch

File: src/components/desktop/desktop.vue (hunk 2 above)
Comment (tajespasarela) [reply src/components/desktop/desktop.vue:15]: I have no idea, we changed the locations definitions too many times. Let's leave it as `none` at the moment. We can change it in whenever is necessary

File: src/components/main.vue [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+          <SpellcheckMessage />
+          <NoResultsMessage />
+        </div>
+      </div>
+
+      <Results />
+
+      <PartialResults />
+
+      <LocationProvider v-if="!$x.noResults" location="no_results">
```
Comment (javieri-empathy) [inline src/components/main.vue:20]:
```suggestion
      <LocationProvider v-if="$x.noResults" location="no_results">
```

**Review by javieri-empathy — APPROVED** [review]

**Review by pmareke — APPROVED** [review]

**Pull request merged**